### PR TITLE
fix(postcss-ordered-values): columns transform returning string instead of the AST

### LINKED ISSUE #927: [Bug] : Using grid/column syntax breaks compilation
**Describe the bug**
When I and others attempt to make a production build (`./node_modules/.bin/next build`) of our Next.js projects, compilation fails with an error, specifically `TypeError: f.charCodeAt is not a function`, seemingly as a result of using CSS grid and column syntax:
- `grid-column: span 2;` (https://github.com/vercel/next.js/issues/15026#issue-654297251)
- `columns: 2 auto;` (https://github.com/vercel/next.js/issues/15026#issuecomment-664985667)
- in my case, I `import '@material/layout-grid/dist/mdc.layout-grid.css';` (https://github.com/vercel/next.js/discussions/14907)

Additional notes from @vukers (https://github.com/Timer/cssnano-preset-simple/issues/6#issue-654279922):
> Here are additional findings:
> - this works when used with the slash syntax: `grid-column: span 2 / 1`
> - this works when used in quotes: `grid-column: "span 2"` (though I'm not sure that's actually valid grid syntax)
> - this works when used with cssnano preset set to "default" instead of "simple"
> - this worked in `next@9.4.5-canary.23`, when "cssnano-simple" was loaded via the `dist/` folder instead of the package.
> - this works when used with `cssnano-preset-simple@1.1.1`
> - this works when used with `cssnano-preset-advanced`

**To Reproduce**
1. Clone https://github.com/4cm4k1/personal-website
2. Run `yarn install`
3. Run `yarn build`

Alternatively, per @vukers again (https://github.com/vercel/next.js/issues/15026#issue-654297251):
> 1. Add offending declaration to any CSS file (global or module) (or use a third-party stylesheet with the offending declaration) in a Next.js project, or theoretically another Webpack/CSSNANO-compiled project
> 2. Run `next build`
> 3. See error

```shell
$ next build


(node:18663) UnhandledPromiseRejectionWarning: TypeError: f.charCodeAt is not a function
    at /Users/4cm4k1/Code/personal-website/static/css/432b36672b3b1c7f82ae.css:583:7
    at B.exports (/Users/4cm4k1/Code/personal-website/node_modules/cssnano-preset-simple/dist/index.js:1:929448)
    at new ValueParser (/Users/4cm4k1/Code/personal-website/node_modules/cssnano-preset-simple/dist/index.js:1:663327)
    at ValueParser (/Users/4cm4k1/Code/personal-website/node_modules/cssnano-preset-simple/dist/index.js:1:663351)
    at normalize (/Users/4cm4k1/Code/personal-website/node_modules/cssnano-preset-simple/dist/index.js:1:181915)
    at /Users/4cm4k1/Code/personal-website/node_modules/cssnano-preset-simple/dist/index.js:1:182403
    at /Users/4cm4k1/Code/personal-website/node_modules/cssnano-simple/node_modules/postcss/lib/container.js:135:18
    at Rule.each (/Users/4cm4k1/Code/personal-website/node_modules/cssnano-simple/node_modules/postcss/lib/container.js:101:16)
    at Rule.walk (/Users/4cm4k1/Code/personal-website/node_modules/cssnano-simple/node_modules/postcss/lib/container.js:131:17)
    at /Users/4cm4k1/Code/personal-website/node_modules/cssnano-simple/node_modules/postcss/lib/container.js:148:24
    at AtRule.each (/Users/4cm4k1/Code/personal-website/node_modules/cssnano-simple/node_modules/postcss/lib/container.js:101:16)
    at AtRule.walk (/Users/4cm4k1/Code/personal-website/node_modules/cssnano-simple/node_modules/postcss/lib/container.js:131:17)
    at /Users/4cm4k1/Code/personal-website/node_modules/cssnano-simple/node_modules/postcss/lib/container.js:148:24
    at AtRule.each (/Users/4cm4k1/Code/personal-website/node_modules/cssnano-simple/node_modules/postcss/lib/container.js:101:16)
    at AtRule.walk (/Users/4cm4k1/Code/personal-website/node_modules/cssnano-simple/node_modules/postcss/lib/container.js:131:17)
    at /Users/4cm4k1/Code/personal-website/node_modules/cssnano-simple/node_modules/postcss/lib/container.js:148:24
    at Root.each (/Users/4cm4k1/Code/personal-website/node_modules/cssnano-simple/node_modules/postcss/lib/container.js:101:16)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:18663) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:18663) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
Creating an optimized production build

Failed to compile.

TypeError: f.charCodeAt is not a function


> Build error occurred
Error: > Build failed because of webpack errors
    at build (/Users/4cm4k1/Code/personal-website/node_modules/next/dist/build/index.js:15:918)
```

**Expected behavior**
Builds successfully.

**Screenshots**
N/A

**Desktop (please complete the following information):**
- CSSNANO Version [e.g. 22]: 4.1.10
- Plugins/presets versions:
- Add the output of `npx envinfo && npm ls cssnano` here:
```shell
  System:
    OS: macOS 11.0
    CPU: (8) x64 Intel(R) Core(TM) i7-3820QM CPU @ 2.70GHz
    Memory: 849.02 MB / 16.00 GB
    Shell: 5.8 - /usr/local/bin/zsh
  Binaries:
    Node: 14.6.0 - /usr/local/bin/node
    Yarn: 1.22.4 - /usr/local/bin/yarn
    npm: 6.14.7 - /usr/local/bin/npm
  Managers:
    Cargo: 1.45.0 - /usr/local/bin/cargo
    CocoaPods: 1.9.3 - /usr/local/bin/pod
    Gradle: 6.5.1 - /usr/local/bin/gradle
    Homebrew: 2.4.9 - /usr/local/bin/brew
    pip3: 20.1.1 - /usr/local/bin/pip3
    RubyGems: 3.1.2 - /usr/local/opt/ruby/bin/gem
  Utilities:
    Bazel: 3.4.1 - /usr/local/bin/bazel
    CMake: 3.18.0 - /usr/local/bin/cmake
    Make: 3.81 - /usr/bin/make
    GCC: 4.2.1 - /usr/bin/gcc
    Git: 2.28.0 - /usr/local/bin/git
    Clang: 1200.0.26.2 - /usr/bin/clang
  Servers:
    Apache: 2.4.43 - /usr/sbin/apachectl
  Virtualization:
    Docker: 19.03.12 - /usr/local/bin/docker
  SDKs:
    iOS SDK:
      Platforms: iOS 14.0, DriverKit 20.0, macOS 11.0, tvOS 14.0, watchOS 7.0
    Android SDK:
      API Levels: 28, 29, 30
      Build Tools: 28.0.3, 29.0.3, 30.0.1
      System Images: android-29 | Google Play Intel x86 Atom_64, android-30 | Google Play Intel x86 Atom_64
  IDEs:
    Nano: 4.9.3 - /usr/local/bin/nano
    VSCode: 1.48.0 - /usr/local/bin/code
    Vim: 8.2 - /usr/bin/vim
    Xcode: 12.0/12A8169g - /usr/bin/xcodebuild
  Languages:
    Bash: 5.0.18 - /usr/local/bin/bash
    Go: 1.14.5 - /usr/local/bin/go
    Java: 1.8.0_262 - /usr/bin/javac
    Perl: 5.28.2 - /usr/bin/perl
    PHP: 7.3.18 - /usr/bin/php
    Protoc: 3.12.3 - /usr/local/bin/protoc
    Python: 3.8.5 - /usr/local/opt/python/libexec/bin/python
    Python3: 3.8.5 - /usr/local/bin/python3
    Ruby: 2.7.1 - /usr/local/opt/ruby/bin/ruby
    Rust: 1.45.0 - /usr/local/bin/rustc
  Databases:
    SQLite: 3.28.0 - /Users/4cm4k1/Library/Android/sdk/platform-tools/sqlite3
  Browsers:
    Chrome Canary: 86.0.4215.0
    Firefox Developer Edition: 79.0
    Safari: 14.0

yarn list v1.22.4
warning Filtering by arguments is deprecated. Please use the pattern option instead.
├─ cssnano-preset-default@4.0.7
├─ cssnano-preset-simple@1.1.4
├─ cssnano-simple@1.0.5
├─ cssnano-util-get-arguments@4.0.0
├─ cssnano-util-get-match@4.0.0
├─ cssnano-util-raw-cache@4.0.1
├─ cssnano-util-same-parent@4.0.1
└─ cssnano@4.1.10
✨  Done in 0.98s.
```

**Additional context**
N/A

x-ref: https://github.com/vercel/next.js/issues/15026
x-ref: https://github.com/Timer/cssnano-preset-simple/issues/6

### PR DESCRIPTION
**Pull request by anikethsaha**
Fixes #927 

`postcss-order-values` when transforming the `columns` was returning ASTs instead of `string`

### CODE REVIEW

File: packages/postcss-normalize-url/src/index.js [hunk 1]
```diff
@@ -46,6 +46,11 @@ function transformNamespace(rule) {
 }
 
 function transformDecl(decl, opts) {
+  // Ref https://github.com/cssnano/cssnano/issues/927
+  if (typeof decl.value !== 'string') {
+    decl.value = decl.value.toString();
+  }
```
Comment (evilebottnawi) [inline packages/postcss-normalize-url/src/index.js:52]: `decl.value` should be always `string`

File: packages/postcss-normalize-string/src/index.js [hunk 2]
```diff
@@ -259,7 +259,14 @@ export default postcss.plugin('postcss-normalize-string', (opts) => {
...
 
-        const newValue = normalize(node[param], preferredQuote);
+        let newValue;
+
+        // Ref https://github.com/cssnano/cssnano/issues/927
+        if (Array.isArray(node[param])) {
+          newValue = normalize(node[param][0], preferredQuote);
+        } else {
+          newValue = normalize(node[param], preferredQuote);
+        }
```
Comment (evilebottnawi) [inline packages/postcss-normalize-string/src/index.js:269]: We should always return `string`

File: packages/postcss-normalize-string/src/index.js [hunk 3]
```diff
@@ -259,7 +259,14 @@ export default postcss.plugin('postcss-normalize-string', (opts) => {
           return;
         }
 
-        const newValue = normalize(node[param], preferredQuote);
+        let newValue;
+
+        // Ref https://github.com/cssnano/cssnano/issues/927
+        if (Array.isArray(node[param])) {
+          newValue = normalize(node[param][0], preferredQuote);
```
Comment (evilebottnawi) [inline packages/postcss-normalize-string/src/index.js:266]: What about `[1]`/`[2]`/etc?

File: packages/postcss-normalize-url/src/index.js (hunk 1 above)
Comment (anikethsaha) [reply packages/postcss-normalize-url/src/index.js:52]: but it is coming as non string

```js
  nodes: [
        { type: 'word', sourceIndex: 0, value: '2' },
        { type: 'space', sourceIndex: 1, value: ' ' },
        { type: 'word', sourceIndex: 2, value: 'auto' }
      ]
```

File: packages/postcss-normalize-string/src/index.js (hunk 3 above)
Comment (anikethsaha) [reply packages/postcss-normalize-string/src/index.js:266]: I have tested a few cases, the array length was always one only. 

I don't know why the value is being converted into an array of the same content.

File: packages/postcss-normalize-string/src/index.js (hunk 2 above)
Comment (anikethsaha) [reply packages/postcss-normalize-string/src/index.js:269]: adding `.toString` ?

File: packages/postcss-normalize-url/src/index.js (hunk 1 above)
Comment (evilebottnawi) [reply packages/postcss-normalize-url/src/index.js:52]: We need investigate why we have array there and should fix it where we put array this, not here

File: packages/postcss-normalize-string/src/index.js (hunk 3 above)
Comment (evilebottnawi) [reply packages/postcss-normalize-string/src/index.js:266]: You need to search why it happens

File: packages/postcss-normalize-string/src/index.js (hunk 2 above)
Comment (evilebottnawi) [reply packages/postcss-normalize-string/src/index.js:269]: Please study the error in detail

**Comment by anikethsaha** [comment]
cc @evilebottnawi fixed the root cause

**Pull request merged**